utf8.h

```diff
@@ -520,6 +526,38 @@ void *utf8ncpy(void *utf8_restrict dst, const void *utf8_restrict src,
   return dst;
 }
 
+void *utf8ndup(const void *src, size_t n) {
+  const char *s = (const char *)src;
+  char *c = 0;
+
+  // figure out how many bytes (including the terminator) we need to copy first
+  size_t bytes = utf8size(src);
```

Comment:
I wonder if there is an issue here, let me try and explain my (perhaps convoluted) thinking:

* We are getting the size of the entire string src - but since we are using one of the utf8n* functions, src might not actually be a null terminated string.
* I think we need to do what utf8size does, but only up to n bytes.

Make sense?

Reply:
Not really convoluted. I think I understand what you are saying. Just pushed that change up in 82afb63